compensations/src/main/java/org/jboss/narayana/compensations/impl/ParticipantImpl.java

```diff
@@ -92,9 +94,10 @@ public void close() throws WrongStateException, SystemException {
             ConfirmationHandler handler = instantiate(confirmationHandler);
             handler.confirm();
 
-            CompensationContext.close(currentTX);
             Thread.currentThread().setContextClassLoader(origClassLoader);
         }
+
+        CompensationContextLifetimeManager.getInstance().participantDestroyed(currentTX);
```

Comment:
Well it's WS-BA participant, so this might be recalled by recovery. However, I'm not sure about the life cycle of the context. To be honest, my first instinct was to add it to the final block, but I removed it later.